### PR TITLE
PR #3

### PR DESCRIPTION
**Pull request by jimmyhurricanedang**
- Added RouteControllerTest for get and delete
- Added a dialog window for adding route
- Added a panto icon 
- Added cards to display the routes
- Set up a default route in backend

### CODE REVIEW

File: backend/src/main/java/de/jimmy/columbus/controller/RouteController.java [hunk 1]
```diff
@@ -30,4 +30,10 @@ public Route addRoute(@RequestBody AddRouteDto dto){
         return routeService.addRoute(dto);
     }
 
+    @DeleteMapping("{name}")
+    public void deleteRoute(@PathVariable String name) {
+        routeService.deleteRouteFromList(name);
+    }
+
```
Comment (NilsMester) [inline backend/src/main/java/de/jimmy/columbus/controller/RouteController.java:37]: Already fixed! ;)

File: backend/src/main/java/de/jimmy/columbus/model/Route.java [hunk 2]
```diff
@@ -20,11 +18,9 @@
 @Document(collection = "routes")
 public class Route {
     @Id
+    private String id;
```
Comment (NilsMester) [inline backend/src/main/java/de/jimmy/columbus/model/Route.java:21]: you don't need the id in the dto.

File: backend/src/main/java/de/jimmy/columbus/service/RouteService.java [hunk 3]
```diff
@@ -33,15 +31,21 @@ public RouteService(RoutesMongoDb routeDb, TimestampUtils timestampUtils){
 
     public Route addRoute(AddRouteDto routeToBeAdded){
         Route routeObjectToBeSaved = Route.builder()
+                .id(routeToBeAdded.getId())
```
Comment (NilsMester) [inline backend/src/main/java/de/jimmy/columbus/service/RouteService.java:34]: no id in the dto. ;)

File: backend/src/main/java/de/jimmy/columbus/service/RouteService.java [hunk 4]
```diff
@@ -33,15 +31,21 @@ public RouteService(RoutesMongoDb routeDb, TimestampUtils timestampUtils){
...
 
         routeDb.save(routeObjectToBeSaved);
-        return (routeObjectToBeSaved);
+        return routeObjectToBeSaved;
 
     }
+
+    public void deleteRouteFromList (String name) {
+        routeDb.deleteById(name);
+    }
```
Comment (NilsMester) [inline backend/src/main/java/de/jimmy/columbus/service/RouteService.java:49]: already fixed. 💪

File: frontend/src/components/RoutesOverview.js [hunk 5]
```diff
@@ -22,10 +23,21 @@ export default function RoutesOverview(){
...
     }
 
-    return(
-        <div>
-            <RouteList routes={routes}/>
+    const deleteRoute = (routeId) => {
+        deleteRouteFromList(routeId).then(() => {
+            setRoutes([
+                ...routes.filter((route) => route.id !== routeId),
+            ])
```
Comment (NilsMester) [inline frontend/src/components/RoutesOverview.js:30]: over here you don't need the array brackets and the spread-operator (...). 

Just 
.then(() => routes.filter((route) => route.id !== routeId)))
works. 

Maybe add a ".catch(console.log);"  at the end.

**Review by NilsMester — COMMENTED** [review]
I just added small comments. Most of we already talked about.

**Comment by JimmyDang95** [comment]
@NilsMester Thank you for reviewing ! Helped me a lot !

**Pull request closed without merging**